parser-native/src/main/java/ru/urururu/sanity/cpp/NativeInstructionParser.java

```diff
@@ -0,0 +1,478 @@
+import ru.urururu.util.Iterables;
+
+import java.util.*;
+
+
+/**
+ * @author <a href="mailto:dmitriy.g.matveev@gmail.com">Dmitry Matveev</a>
+ */
+@Component
+public class NativeInstructionParser extends InstructionParser<SWIGTYPE_p_LLVMOpaqueType,
+        SWIGTYPE_p_LLVMOpaqueValue, SWIGTYPE_p_LLVMOpaqueValue, SWIGTYPE_p_LLVMOpaqueBasicBlock, NativeCfgBuildingCtx> {
+    @Autowired
+    ConstCache constants;
+
+    private Map<LLVMOpcode, OpcodeParser> opcodeParsers;
+
+    private OpcodeParser defaultParser = new AbstractParser() {
+        @Override
+        public Set<LLVMOpcode> getOpcodes() {
+            return Collections.emptySet();
+        }
+    };
+
+    public NativeInstructionParser() {
+        this.opcodeParsers = FinalMap.createHashMap();
+
+        List<OpcodeParser> parsersOtu = new ArrayList<>();
+        parsersOtu.add(new StoreParser());
+        parsersOtu.add(new LoadParser());
+        parsersOtu.add(new RetParser());
+        parsersOtu.add(new AllocaParser());
+        parsersOtu.add(new GetElementPtrParser());
+        parsersOtu.add(new ExtractValueParser());
+        parsersOtu.add(new CallParser());
+        parsersOtu.add(new PhiParser());
+        parsersOtu.add(new BrParser());
+        parsersOtu.add(new SwitchParser());
+        parsersOtu.add(new BinaryOperationParser());
+        parsersOtu.add(new BitCastParser());
+        parsersOtu.add(new CastParser());
+
+        parsersOtu.add(new ICmpParser());
+        parsersOtu.add(new FCmpParser());
+
+        for (OpcodeParser parser : parsersOtu) {
+            for (LLVMOpcode opcode : parser.getOpcodes()) {
+                opcodeParsers.put(opcode, parser);
+            }
+        }
+    }
+
+    protected Cfe doParse(NativeCfgBuildingCtx ctx, SWIGTYPE_p_LLVMOpaqueValue instruction) {
+        OpcodeParser parser = opcodeParsers.getOrDefault(bitreader.LLVMGetInstructionOpcode(instruction), defaultParser);
+
+        return parser.parse(ctx, instruction);
+    }
+
+    public RValue parseValue(NativeCfgBuildingCtx ctx, SWIGTYPE_p_LLVMOpaqueValue instruction) {
+        OpcodeParser parser = opcodeParsers.getOrDefault(bitreader.LLVMGetInstructionOpcode(instruction), defaultParser);
+        return parser.parseValue(ctx, instruction);
+    }
+
+    public RValue parseConst(NativeCfgBuildingCtx ctx, SWIGTYPE_p_LLVMOpaqueValue constant) {
+        OpcodeParser parser = opcodeParsers.getOrDefault(bitreader.LLVMGetConstOpcode(constant), defaultParser);
+        return parser.parseConst(ctx, constant);
+    }
+
+    private interface OpcodeParser {
+        Set<LLVMOpcode> getOpcodes();
+
+        Cfe parse(NativeCfgBuildingCtx ctx, SWIGTYPE_p_LLVMOpaqueValue instruction);
+
+        RValue parseValue(NativeCfgBuildingCtx ctx, SWIGTYPE_p_LLVMOpaqueValue instruction);
+
+        RValue parseConst(NativeCfgBuildingCtx ctx, SWIGTYPE_p_LLVMOpaqueValue constant);
+    }
+
+    private static abstract class AbstractParser implements OpcodeParser {
```

Comment:
![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png 'Severity: MINOR') Reorder the modifiers to comply with the Java Language Specification. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonarqube.com/coding_rules#rule_key=squid%3AModifiersOrderCheck)